### ISSUE TITLE
Using lodash-es

### ISSUE THREAD

**Issue by KidkArolis** [gaearon/react-hot-loader#1269]
If you configure a lodash alias in webpack:

```
config.resolve.alias['lodash'] = path.dirname(require.resolve('lodash-es'))
```

RHL fails with:

```
Uncaught TypeError: merge is not a function
    at updateProxyById (react-hot-loader.development.js:1089)
    at Object.register (react-hot-loader.development.js:2533)
```

Because `const merge = require('lodash-es/merge')` results in `{ default: merge }`.
Haven't figured out how to workaround this yet.

**Comment by KidkArolis** [comment]
One suggestion would be to ship an esm version of the library linked to under "module" entry.

**Comment by KidkArolis** [comment]
For now, this workaround works:

```
if (!options.production) {
  config.resolve.alias['lodash/merge'] = path.dirname(require.resolve('lodash/merge'))
}
config.resolve.alias['lodash'] = path.dirname(require.resolve('lodash-es'))
 ```

**Comment by KidkArolis** [comment]
Would you accept a PR that removes the use of lodash and replaces lodash/merge with something inline.

**Comment by theKashey** [comment]
With pleasure :)

**Comment by msudgh** [comment]
I'm wondering the following error is related to this issue:
```bash

ERROR in ./node_modules/react-hot-loader/dist/react-hot-loader.development.js
Module not found: Error: Can't resolve 'lodash/merge' in '/Users/mcrunix/w/phizog/node_modules/react-hot-loader/dist'
 @ ./node_modules/react-hot-loader/dist/react-hot-loader.development.js 1057:12-35
 @ ./node_modules/react-hot-loader/patch.js
 @ multi react-hot-loader/patch webpack-hot-middleware/client?path=http://localhost:3000/__webpack_hmr&reload=true ./app/index
```

**Comment by theKashey** [comment]
The best solution would be to get rid of `merge`. It's not actually required.

**Comment by msudgh** [comment]
I'm not familiar with the logic of RHL proxy as the following but in case of merging objects you have spreading operator or import merge lib in es6 way `import merge from 'lodash/merge`.

https://github.com/gaearon/react-hot-loader/blob/ddb0a8af6bc11c453b84d61544aadaea97fb69e8/src/reconciler/proxies.js#L59

**Comment by theKashey** [comment]
It's more about - this code is really not needed anymore.

**Comment by msudgh** [comment]
Just another point, I feel it'd be great to consider a stable tag to not release all versions on NPM. Every time i tried to run my project for a 1-2 month period i got an error which related to react-hot-loader.

**Comment by theKashey** [comment]
Should be fixed in 4.12.4

> @msudgh - there is no such thing as "stable" here. I always have to address compatibility issues with the rest of the ecosystem, like this one. The second part of problem is "I", not "We", which hugely affects testing scope.
